esi_leap/common/idp/__init__.py

```diff
@@ -0,0 +1,8 @@
+import os
+
+from esi_leap.common.idp import keystoneIDP, dummyIDP
+
+if os.environ.get('ESI_DEBUG', '') == 'True':
```

Comment:
If we're going to use an environment variable in the short term, I think it needs to be something more specific than `ESI_DEBUG` (because (a) this isn't really a debugging facility, and (b) `ESI_DEBUG` seems a bit broad and it would be tempting to make it do *other* things related to debugging as well). Maybe `ESI_USE_DUMMY_IDP` instead.

I think for the final version of this PR we want to introduce a new config option to control IDP selection, and it should allow us to provide an arbitrary fully qualified class name. Something like:

```
[esi-leap]
idp_plugin=esi_leap.common.idp.keystoneIDP
```
For testing, we should be able to pass in an arbitrary class instance that supports the necessary idp API. This would allow us to perform failure injection and other sorts of tests without needing to monkey patch the existing code.

Reply:
Also, instead of explicitly testing for the value `True`, consider either (a) simply testing if the value is set:

```
if os.environ.get('ESI_DEBUG') is not None:
```

Or (b) be a little more flexible in what you accept as truthy values:

```
if os.environ.get('ESI_DEBUG', '').lower() in ['true', '1']:
```